phable_cli/cli.py

```diff
@@ -59,59 +59,51 @@ def show_task(task_id: int, format: str = "plain"):
-            owner = client.show_user(phid=owner_id)["fields"]["username"]
-        else:
-            owner = "Unassigned"
-        if project_ids := task["attachments"]["projects"]["projectPHIDs"]:
-            tags = [
-                (
-                    f"{project['fields']['parent']['name']} - {project['fields']['name']}"
-                    if project["fields"]["parent"]
-                    else project["fields"]["name"]
-                )
-                for project in client.show_projects(phids=project_ids)
-            ]
-        else:
-            tags = []
-        subtasks = client.find_subtasks(parent_id=task_id)
-        task["subtasks"] = subtasks
-        parent = client.find_parent_task(subtask_id=task_id)
-        task["parent"] = parent
-        if format == "json":
-            click.echo(json.dumps(task))
-        else:
-            parent_str = (
-                f"{Task.from_int(parent['id'])} - {parent['fields']['name']}"
-                if parent
-                else ""
-            )
-            click.echo(f"URL: {client.base_url}/{Task.from_int(task_id)}")
-            click.echo(f"Task: {Task.from_int(task_id)}")
-            click.echo(f"Title: {task['fields']['name']}")
-            click.echo(f"Author: {author['fields']['username']}")
-            click.echo(f"Owner: {owner}")
-            click.echo(f"Tags: {', '.join(tags)}")
-            click.echo(f"Status: {task['fields']['status']['name']}")
-            click.echo(f"Priority: {task['fields']['priority']['name']}")
-            click.echo(f"Description: {task['fields']['description']['raw']}")
-            click.echo(f"Parent: {parent_str}")
-            click.echo("Subtasks:")
-            if subtasks:
-                for subtask in subtasks:
-                    status = f"{'[x]' if subtask['fields']['status']['value'] == 'resolved' else '[ ]'}"
-                    if subtask_owner_id := subtask["fields"]["ownerPHID"]:
-                        owner = client.show_user(subtask_owner_id)["fields"]["username"]
-                    else:
-                        owner = ""
-                    click.echo(
-                        f"{status} - {Task.from_int(subtask['id'])} - @{owner:<10} - {subtask['fields']['name']}"
-                    )
+        task = client.enrich_task(task, with_author_owner=True, with_tags=True, with_subtasks=True, with_parent=True)
+        echo_task(click.echo, format, task)
     else:
         click.echo(f"Task {Task.from_int(task_id)} not found")
 
 
+def echo_task(echo: Callable[[str], None], format: str, task: dict[str, Any]) -> None:
+    """Print a task.
+
+    Print a task in a text or json format. The task needs to be enriched first.
+
+    To generalize the implementation and not couple it to the click library,
+    the user must pass an `echo` function that will be used to print the task.
+    """
+    if format == "json":
+        echo(json.dumps(task))
+    else:
+        echo("="*50)
```

Comment:
Can we move this to the outer `for` loop, so that it does not impact the output of `phable show` ?

Reply:
Or maybe add a boolean `separator` flag, and only print this if `separator == True` ? I'll implement some appropriate display logic later on.